### ISSUE TITLE
race condition with forced option

### ISSUE THREAD

**Issue by artwells** [artwells/meteor-accounts-guest#12]
"Often, the code in accounts-guest-client.js from line 48 kicks in before I had a chance to set AccountsGuest.forced to false."

**Comment by prinzdezibel** [comment]
Have you already had the possibility to check this bug?

**Comment by artwells** [comment]
This does seem to be fixed.  In the README, I've add that the options should be set in the server at start up.

**Comment by Levino** [comment]
Has this been fixed? If I run it in a callback passed to "Meteor.startup()" it is too late.

**Comment by artwells** [comment]
I'll take a look

**Comment by Levino** [comment]
I threw it in somewhere in the code in the "lib" folder. That "did the trick". I think this should be done in the settings.json file and not in a global variable.

**Comment by Levino** [comment]
https://github.com/artwells/meteor-accounts-guest/blob/master/accounts-guest-client.js#L35

```
Meteor.loginVisitor = function (email, callback) {
    AccountsGuest.forced = true;
    if (!Meteor.userId()) {
        Meteor.call('createGuest', email, function (error, result) {
            if (error) {
                console.log('Error in creating Guest ' + error);
                return callback && callback(error);
            }

            /* if a simple "true" is returned, we are in a disabled mode */
            if(result === true) return callback && callback();

            Meteor.loginWithPassword(result.email, result.password, function(error) {
                if(error) {
                    console.log('rrError logging in ' + error);
                    callback && callback(error);
                } else {
                    callback && callback();
                }
            });
        });
    }
}
```

Setting the flag to force the relogin is a bug here. I want a button for the user to sign in with a demo account. But I also want him to be able to log out again. That is impossible when the flag is set from false to true.

Background: I am also using useraccounts like @buduguru in #17 . I want to solve the problem in this way: User can use a demo account. He gets a button "Sign Up". On click I sign him out and throw him to the sign-up page. Unfortunately right now he gets signed it automatically.

I do not see a use case for setting this "environment variable" here at all. The line should be deleted.

**Comment by artwells** [comment]
fixed by Levino in 0.1.11

**Comment by artwells** [comment]
closed in error
